routes/resources.js

```diff
@@ -32,6 +32,34 @@ module.exports = function (app, model, Implementation, integrator, opts) {
         } else if (_.isArray(field.type)) {
           record[field.field] = [];
         }
+      } else if(field.reference) {
+        console.log('coucou',field);
+        // Inject Smart Field in BelongsTo Fields
+        var schemaName = field.reference.split('.')[0];
+        var referenceFieldSchema = Schemas.schemas[schemaName];
+        if (referenceFieldSchema) {
+          _.each(referenceFieldSchema.fields, function (smartFieldObject) {
+            var valueSmartField;
+            if(smartFieldObject.isVirtual &&  (smartFieldObject.get ||
+              smartFieldObject.value)) {
+              if (smartFieldObject.value) {
+                logger.warn('DEPRECATION WARNING: In your ' + field.field +
+                  ' Collection, Smart Field value method is deprecated. ' +
+                  'Please use get method instead. ');
+                valueSmartField = smartFieldObject.value(record[field.field]);
+              } else {
+                valueSmartField = smartFieldObject.get(record[field.field]);
+              }
+              if (valueSmartField && _.isFunction(valueSmartField.then)) {
+                return valueSmartField.then(function (value) {
+                  record[field.field][smartFieldObject.field] = value;
+                });
+              } else {
+                record[field.field][smartFieldObject.field] = valueSmartField;
+              }
```

Comment:
The 18 previous lines are duplicated from the existing code. You need to factorise it: define a function that will be used by both cases (`if` and `elsif` cases)